django/contrib/auth/models.py

```diff
@@ -142,8 +143,12 @@ def _create_user(self, username, email, password, **extra_fields):
         if not username:
             raise ValueError('The given username must be set')
         email = self.normalize_email(email)
-        username = self.model.normalize_username(username)
-        user = self.model(username=username, email=email, **extra_fields)
+        # Lookup the real model class from the global app registry so this
+        # manager method can be used in migrations. This is fine because
+        # managers are by definition working on the real model.
+        UserModel = apps.get_model(self.model._meta.app_label, self.model._meta.object_name)
+        username = UserModel.normalize_username(username)
+        user = UserModel(username=username, email=email, **extra_fields)
```

Comment:
What about using the global user model's `normalize_username` method while returning an instance of `self.model`?

```python
GlobalUserModel = apps.get_model(self.model._meta.app_label, self.model._meta.object_name)
username = GlobalUserModel.normalize_username(username)
password = GlobalUserModel.create_password(password)
user = UserModel(username=username, email=email, **extra_fields)
user.password = password
user.save(using=self._db)
return user
```